### DISCUSSION TITLE
How to fix one volume?

### DISCUSSION THREAD

**Discussion by aoberest** [seaweedfs/seaweedfs#2612] [Q&A]
Hello.

After an unexpected server shutdown, I got a bad volume. What are the options to fix it?

Right now I'm getting a lot of errors on weed-volume
weed-volume[110162]: E0125 08:20:29 10162 needle_read_write.go:220] entry not found1: offset 21534273128 found id 31663333346431366165623630303632 size 1972662651, expected size 173225
or
weed-volume[110162]: E0125 08:26:32 10162 needle_read_write.go:208] /var/lib/seaweedfs/volume/bucket_22.dat read 0 dataSize 262952 offset 56105459192 fileSize 21759449880: EOF

**Comment by chrislusf** [comment]
use command "weed fix"

**Reply by aoberest** [reply]
Should we stop the weed volume service on this node?

**Reply by chrislusf** [reply]
Yes

**Reply by aoberest** [reply]
We have replication 001.

I made a weed fix.
There were fewer mistakes, all did not go away.

```
weed-volume[83461]: E0125 09:24:59 83461 needle_read_write.go:208] /var/lib/seaweedfs/volume/bucket_2.dat read 0 dataSize 402424 offset 21417636128 fileSize 13836688408: EOF
weed-volume[83461]: E0125 09:25:01 83461 needle_read_write.go:208] /var/lib/seaweedfs/volume/bucket_2.dat read 0 dataSize 378840 offset 16392531664 fileSize 13836688408: EOF
weed-volume[83461]: E0125 09:25:04 83461 needle_read_write.go:208] /var/lib/seaweedfs/volume/bucket_2.dat read 0 dataSize 378840 offset 16392531664 fileSize 13836688408: EOF
weed-volume[83461]: E0125 09:25:05 83461 needle_read_write.go:208] /var/lib/seaweedfs/volume/bucket_2.dat read 0 dataSize 173248 offset 21489005712 fileSize 13836688408: EOF
weed-volume[83461]: E0125 09:25:07 83461 needle_read_write.go:208] /var/lib/seaweedfs/volume/bucket_2.dat read 0 dataSize 173248 offset 21489005712 fileSize 13836688408: EOF
E0125 09:25:09 83461 needle_read_write.go:208] /var/lib/seaweedfs/volume/bucket_2.dat read 0 dataSize 378840 offset 16392531664 fileSize 13836688408: EOF
````

tell me please, How to fix this error ?

**Reply by chrislusf** [reply]
Since you have another copy, you can just copy over the good replica.

**Comment by Andrmist** [comment]
I have these issues too on 4.23 version, but I don't use replication. Is there any chances to restore data?
